modules/lib/src/main/resources/assets/js/app/wizard/page/LiveFormPanel.ts

```diff
@@ -987,4 +1009,29 @@ export class LiveFormPanel
     setSaveEnabled(enabled: boolean): void {
         this.saveAction.setEnabled(enabled);
     }
+
+    private updateRenderingState(): void {
+        if (this.isRenderable) {
+            this.clearErrorMissingApps();
+        } else {
+            if (this.hasPage) {
+                if (this.hasMissingApps) { // not renderable due to missing apps
+                    this.setErrorMissingApps();
+                } else { // not renderable due to rendering error
+                    this.clearErrorMissingApps();
+
+                    if (!this.placeholder) {
+                        this.createLiveEditPagePlaceholder();
+                    }
+
+                    this.frameContainer?.hide();
+                    this.placeholder.setPageIsNotRenderableMode();
+                    this.placeholder.show();
+                }
+            } else { // no page
+                this.clearErrorMissingApps();
+                this.placeholder?.setHasControllersMode(this.hasAvailableControllers);
+            }
+        }
+    }
```

Comment:
```suggestion
        if (this.isRenderable) {
            this.clearErrorMissingApps();
            return;
        }
        
        if (!this.hasPage) {  // no page
            this.clearErrorMissingApps();
            this.placeholder?.setHasControllersMode(this.hasAvailableControllers);
            return;
        }

        if (this.hasMissingApps) { // not renderable due to missing apps
          this.setErrorMissingApps();
          return;
        }
                
        // not renderable due to rendering error
        this.clearErrorMissingApps();

        if (!this.placeholder) {
            this.createLiveEditPagePlaceholder();
        }

        this.frameContainer?.hide();
        this.placeholder.setPageIsNotRenderableMode();
        this.placeholder.show();
    }
```

Reply:
Can probably be refactored as well, since `this.clearErrorMissingApps()` is called in 3 out of 4 cases, so there's no point in repeating it.

Reply:
Functionality now works fine, btw. Great job